src/resources/raw.ts

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 Cognite AS
+
+import { AxiosInstance } from 'axios';
+import { CogniteAsyncIterator } from '../autoPagination';
+import { rawRequest } from '../axiosWrappers';
+import { MetadataMap } from '../metadata';
+import {
+  generateCreateEndpoint,
+  generateDeleteEndpoint,
+  generateListEndpoint,
+  generateRetrieveSingleEndpoint,
+} from '../standardMethods';
+import {
+  ItemsResponse,
+  ListRawDatabases,
+  ListRawRows,
+  ListRawTables,
+  RawDB,
+  RawDBRow,
+  RawDBRowKey,
+  RawDBTable,
+} from '../types/types';
+import { projectUrl } from '../utils';
+
+export interface RawAPI {
+  /**
+   * [List databases](https://doc.cognitedata.com/api/v1/#operation/getDBs)
+   *
+   * ```js
+   * const databases = await client.raw.listDatabases();
+   * ```
+   */
+  listDatabases: (query?: ListRawDatabases) => CogniteAsyncIterator<RawDB>;
+
+  /**
+   * [Create databases](https://doc.cognitedata.com/api/v1/#operation/createDBs)
+   *
+   * ```js
+   * const databases = await client.raw.createDatabases([{ name: 'My company' }]);
+   * ```
+   */
+  createDatabases: (items: RawDB[]) => Promise<RawDB[]>;
+
+  /**
+   * [Delete databases](https://doc.cognitedata.com/api/v1/#operation/deleteDBs)
+   *
+   * ```js
+   * await client.raw.deleteDatabases([{ name: 'My company' }]);
+   */
+  deleteDatabases: (items: RawDB[]) => Promise<{}>;
+
+  /**
+   * [List tables in database](https://doc.cognitedata.com/api/v1/#operation/getTables)
+   *
+   * ```js
+   * const tables = await client.raw.listTables('My company');
+   * ```
+   */
+  listTables: (
+    databaseName: string,
+    query?: ListRawTables
+  ) => CogniteAsyncIterator<RawDBTable>;
+
+  /**
+   * [Create tables in a database](https://doc.cognitedata.com/api/v1/#operation/createTables)
+   *
+   * ```js
+   * const tables = await client.raw.createTables('My company', [{ name: 'Customers' }]);
+   * ```
+   */
+  createTables: (
+    databaseName: string,
+    items: RawDBTable[],
+    // tslint:disable-next-line:bool-param-default
```

Comment:
why did you disable it here?
doesn't `false` means the same for us as `undefined`?
(I mean we won't create db in both cases, so it might be ok to set `default=false`)

Reply:
I tried both:
- `ensureParent: boolean = false` (it fails since you can also have an initialiser in the function declaration and not in typescript declaration.
- `ensureParent: boolean | undefined` (but ts-lint also complained about this one). So thought it was a bug in ts-lint.

You can try out yourself on that branch, but it was always complaining